### DISCUSSION TITLE
How to make bot read the text after bot asked user to write something?

### DISCUSSION THREAD

**Discussion by msvetofor** [def-studio/telegraph#436] [Q&A] [answered]
Example:
bot sends message: "write something"
user writes something
bot uses message from user

I know about handleChatMessage, but this works after every non-command message. I need this to work only when bot asks user to write.

**Comment by fabio-ivona** [comment, marked as answer]
Hi, there's not a straight way to do this, you should track the conversation status (maybe using telegraph storage) and retrieve it when you receive a message from that user

**Comment by FrontEnd07** [comment]
```
public function search()
   {
      $this->chat->message('message:')->reply($this->messageId)->send();
      $this->chat->storage()->set('tracker', true);
   }
```
```
protected function handleChatMessage(Stringable $text): void
   {
      if ($this->chat->storage()->get("tracker")) {
         $this->chat->storage()->set("tracker", null);
         $this->chat->message($text)->send();
       } 
   }
```

**Reply by man0so1** [reply]
Thank you very much for this answer!
